daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -700,6 +701,30 @@ trait ElementBaseGrammarMixin
       ConvertZonedCombinator(this, stringValue, textZonedConverter)
     }
 
+  /**
+   * True if the encoding is known to be an EBCDIC one, as in the encoding is not  a runtime expression
+   * and it's some ebcdic flavor. If it's any ascii flavor or a runtime expression this is false.
+   */
+  lazy val isKnownEBCDICEncoding: Boolean = {
+    charsetEv.optConstant
+      .map { bcs: BitsCharset =>
+        val nom = bcs.name.toUpperCase()
+        val res = (nom == "IBM037") ||
+          nom.startsWith("EBCDIC")
+        res
+      }
+      .getOrElse(false)
+  }
+
+  /**
+   * Avoids requesting the textZonedSignStyle property if we know the encoding
+   * is an EBCDIC flavor.
+   */
+  lazy val optTextZonedSignStyle = {
+    if (isKnownEBCDICEncoding) None
+    else Some(textZonedSignStyle)
+  }
```

Comment:
This doesn't do the right thing if the charset is non-constant be evaluates to EBCIDC at runtime. Is that something we care about? Does it make sense to convert this to a evaluatable?

Reply:
But if the charset evaluates to EBCDIC at runtime we'll still use textZonedSignStyle. So in that case we still need the property at compile time (like you have) but we need to ignore it at runtime.

So maybe the parser/unparsers need to have the charsetEv passed in, and if at runtime if it's non-constant and evaluates to EBCDIC then it ignores `optTextZonedSignStyle`?

Reply:
Correct. That is what happens. So this optional thing is just about not reqiring what is an ascii-oriented property at all in the case where it is known for certain to be EBCDIC.